pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -321,12 +310,12 @@ spec:
 		"Normal Started",
 		"Normal Running Tasks Completed: 0",
 	}
-	reconciledRun, clients := prt.reconcileRun("foo", "test-pipeline-run-success", wantEvents, false)
+	reconciledRun, clients := prt.reconcileRun(namespace, prName, wantEvents, false)
 
 	// Check that the expected TaskRun was created
-	actual := getTaskRunCreations(t, clients.Pipeline.Actions(), 2)[0]
+	actual := getTaskRunsForPipelineRun(prt.TestAssets.Ctx, t, clients, namespace, prName, 2)[0]
```

Comment:
It looks like this relies on the ordering in which the taskruns are returned, which isn't guaranteed. For this test, it might be better to fetch the 2 taskruns by name. Maybe the function `getTaskRunsForPipelineRun` could return a map of TaskRun name -> TaskRun, instead of just a slice of TaskRuns?

Reply:
> It looks like this relies on the ordering in which the taskruns are returned, which isn't guaranteed. For this test, it might be better to fetch the 2 taskruns by name. Maybe the function `getTaskRunsForPipelineRun` could return a map of TaskRun name -> TaskRun, instead of just a slice of TaskRuns?

How about passing a name to getTaskRunsForPipelineRun and just returning one TaskRun？It looks like currently we need one TaskRun only in all the test cases.